test/framework/src/main/java/org/elasticsearch/test/AnnotationTestOrdering.java

```diff
@@ -25,10 +26,11 @@
 public class AnnotationTestOrdering implements Comparator<TestMethodAndParams> {
     @Override
     public int compare(TestMethodAndParams o1, TestMethodAndParams o2) {
-        return Integer.compare(
-            o1.getTestMethod().getAnnotation(Order.class).value(),
-            o2.getTestMethod().getAnnotation(Order.class).value()
-        );
+        return Integer.compare(getOrderValue(o1.getTestMethod()), getOrderValue(o2.getTestMethod()));
+    }
+
+    private int getOrderValue(Method method) {
+        return method.isAnnotationPresent(Order.class) ? method.getAnnotation(Order.class).value() : Integer.MAX_VALUE;
     }
```

Comment:
This maybe controversial. It essentially allows annotating `Order` for a subset of tests instead of all the tests. Missing Order annotation defaults to Integer.MAX_VALUE, i.e. they run last. I am trying to avoid annotating every method in `FullClusterRestartIT`. It is really needed for just one method. The rest can run in whatever order. I also considered splitting the single method into a different test class. But `CoreFullClusterRestartIT` extends `FullClusterRestartIT` which means we need yet another test class to extends the new test class. Seems not worth it. So I went with this approach.

Reply:
Nevermind. I just noticed the full-cluster-restart package has its own `FullClusterRestartTestOrdering` annotation. So I have moved the changes into it which should be less of an issue. The tests also need to respect the ordering determined by old and new cluster as well.